action.yaml

```diff
@@ -183,9 +183,12 @@ runs:
       shell: bash
       run: |
         pip3 install pyyaml==6.0.1
+        REPO_INFO=${{ github.repository }}
+        REPO_NAME=$(echo "$REPO_INFO" | cut -d'/' -f2)
 
         python3 ${GITHUB_ACTION_PATH}/citadel/citadel.py \
-          "${GITHUB_WORKSPACE}/citadel-config.yaml" "${{ env.PROJECT_PATH }}" "$GITHUB_WORKFLOW_REF" "$RUNNER_TEMP"
+          "${GITHUB_WORKSPACE}/citadel-config.yaml" "${{ env.PROJECT_PATH }}" "$GITHUB_WORKFLOW_REF" "$RUNNER_TEMP" \
+          "${{ env.CODEDX_BRANCH }}" "${{ github.repository_owner }}" "$REPO_NAME,${{ github.ref_name }}" "${{ env.PLUGIN_NAMES }}" "${{ env.ARTIFACT_URLS }}"
```

Comment:
Maybe the `repo_path_includes` can be retrieved from the change detection? Are are these variables suppose to be populated?

Can't the manifest file be leveraged for the `artifacts_urls` somehow?

Reply:
If someone builds from the root level, the change detection will have "." value instead of plugin names. I think it's better to fetch the URL and plugin names using the build number, so that we do not need to rely on the manifest file

Reply:
> I think it's better to fetch the URL and plugin names using the build number

I don't understand that statement. How does the build number help n this case?

Reply:
The artifact is being deployed using the build number, and we are fetching all artifact URL links for all plugins that were built, along with their names.

![image](https://github.com/lumada-common-services/gh-composite-actions/assets/112469025/2a4d961a-0e66-4fdb-b2fb-137667965023)

For example, here we run the build and deploy steps, and everything is getting deployed inside the '347' (i.e., Build number) folder under 'pdi-plugins-ee.' Therefore, we are using the build number to fetch the URLs and plugins that were deployed using [this](https://github.com/pentaho/actions-common/blob/c936677177ce41c61e832d195c8179aa9cd74b37/.github/workflows/merge.yml#L261)